pkg/filesystem/virtual/configuration/fuse_mount_enabled.go

```diff
@@ -94,5 +94,11 @@ func (m *fuseMount) Expose(terminationGroup program.Group, rootDirectory virtual
 	); err != nil {
 		return util.StatusWrap(err, "Failed to set Linux Backing Device Info tunables")
 	}
+
+	// Access to call mount() is no longer needed now that FUSE has been initialized.
```

Comment:
The downside of this approach is that it doesn't compose well. For example, you can set up multiple build directories. In fact, for FUSE based workers it's actually not uncommon to do that. That way each worker thread is only capable of seeing its own input files. Also, if you refactor your worker config to not use FUSE at all, you'll never end up calling this.

I think we need to introduce some global API (pkg/global?) that allows us to specify 'leases' on capabilities, and have individual subsystems drop these leases. If the number of leases drops to zero, we can actually call `prctl()` to drop them.

Reply:
I am fairly certain we need it for chroot runners too. It is high time I dust that off. But I really appreciate the pledge mentality. Would be very interesting to see how it can be done within Buildbarn.